### PR TITLE
remaining IPv6 fixes for microhttpd web server

### PR DESCRIPTION
**Pull request by wanxewoj**
Renamed `config->gw_address` to `config->gw_ip` as discussed in https://github.com/nodogsplash/nodogsplash/issues/38.
Fixed too small buffer for IPv6 in `our_host` variable in `is_foreign_hosts()` and `is_splashpage()`

### CODE REVIEW

**Comment by bluewavenet** [comment]
@wanxewoj 
Great progress :-)
Can you resubmit this PR against the new v4_Development branch please. This will prevent clashes with ongoing work on v3 and make it easier all round.
I have not looked in detail at the changes, but don't forget consideration for both BinAuth and FAS functionality is also required.

**Comment by mwarning** [comment]
The changes look good. As @bluewavenet said, let's create the PR for merge into v4_Development. We will merge it from there to master after other work on master is finished.

File: src/http_microhttpd.c [hunk 1]
```diff
@@ -498,7 +498,7 @@ static int authenticated(struct MHD_Connection *connection,
 
 	if (check_authdir_match(url, config->denydir)) {
 		auth_client_deauth(client->id, "client_deauth");
-		snprintf(redirect_to_us, 128, "http://%s:%u/", config->gw_address, config->gw_port);
+		snprintf(redirect_to_us, 128, "http://%s/", config->gw_address);
```
Comment (mwarning) [inline src/http_microhttpd.c]: `128` could be replaced with `sizeof(redirect_to_us)`. But it is not important for this PR.

**Comment by bluewavenet** [comment]
Having sat down with a beer and read ALL the source files :-D  Firewall Rulesets are going to need some attention.

**Comment by mwarning** [comment]
corresponding xkcd comic: https://www.xkcd.com/323/ :P

**Comment by wanxewoj** [comment]
> Can you resubmit this PR against the new v4_Development branch please.

Done

> 128 could be replaced with sizeof(redirect_to_us). 

Fixed as well.

**Pull request closed without merging**